cli: enforce inclusion of new crdb_internal tables in debug zip

tbg opened the pull request:
This is achieved via a unit test. I went through all of the missing
tables and made judgement calls on whether they needed to be included.

This resulted in the inclusion of some obvious earlier omissions.

I also went through the crdb_internal tables and added a naming
convention and tried to add TODOs to all tables violating it.
It's a bad idea to rename any of these; we should probably alias
them to their desired name and leave the old name valid, somehow
without listing it in `SHOW TABLES FROM crdb_internal`.

Release note (bug fix): Data that was previously omitted from `debug
zip` is now included.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/36281)
<!-- Reviewable:end -->

andreimatei commented:
LGTM

tbg committed:
cli: enforce inclusion of new crdb_internal tables in debug zip

This is achieved via a unit test. I went through all of the missing
tables and made judgement calls on whether they needed to be included.

This resulted in the inclusion of some obvious earlier omissions.

I also went through the crdb_internal tables and added a naming
convention and tried to add TODOs to all methods violating it.
It's a bad idea to rename any of these; we should probably alias
them to their desired name and leave the old name valid, somehow
without listing it in `SHOW TABLES FROM crdb_internal`.

Release note (bug fix): Data that was previously omitted from `debug
zip` is now included.

tbg commented:
TFTR!

bors r=andreimatei
